writer.go

```diff
@@ -192,23 +192,35 @@ func (w *GenericWriter[T]) Reset(output io.Writer) {
 	w.base.Reset(output)
 }
 
-func (w *GenericWriter[T]) Write(rows []T) (int, error) {
-	return w.base.writer.writeRows(len(rows), func(i, j int) (int, error) {
-		n, err := w.write(w, rows[i:j:j])
-		if err != nil {
-			return n, err
-		}
+func (w *GenericWriter[T]) Write(rows []T) (written int, err error) {
+	var n int
+	for len(rows) > 0 {
+		n, err = w.base.writer.currentRowGroup.writeRows(len(rows), func(i, j int) (int, error) {
```

Comment:
Let's capture `w.base.writer.currentRowGroup` at the beginning of the function to reduce memory loads in the inner loops.

Reply:
Done in latest commit.